### PR TITLE
Din 2814 update for pg15

### PR DESCRIPTION
**Pull request by notoriousR-O-B**

### CODE REVIEW

File: trunkfingerprint--1.2.0.sql [hunk 1]
```diff
@@ -80,7 +78,8 @@ begin
                                  when attname like '%server'    then '(select srvname from pg_foreign_server where oid = foo.' || attname || ')'
                                  when attname like '%fdw'       then '(select fdwname from pg_foreign_data_wrapper where oid = foo.' || attname || ')'
                                  when attname like '%lang'      then '(select lanname from pg_language where oid = foo.' || attname || ')'
-                                 when attname like '%constraint'then '(select conname from pg_constraint where oid = foo.' || attname || ')'
+                                 when attname like '%constraint'
+                                   or attname like '%conparentid' then '(select conname from pg_constraint where oid = foo.' || attname || ')'
```
Comment (bashtanov) [inline trunkfingerprint--1.2.0.sql:82]: we probably need to output the table as well, similar to parent triggers

File: trunkfingerprint--1.2.0.sql [hunk 2]
```diff
@@ -90,6 +89,13 @@ begin
                                                                 then '(select oprname from pg_operator where oid = foo.' || attname || ')'
                                  when attname like '%opc'       then '(select opcname from pg_opclass where oid = foo.' || attname || ')'
                                  when attname like '%tablespace'then '(select spcname from pg_tablespace where oid = foo.' || attname || ')'
+                                 when attname like '%partdefid' then '(select relname from pg_class where oid = foo.' || attname || ')'
```
Comment (bashtanov) [inline trunkfingerprint--1.2.0.sql:92]: We should rather add it to those below at line 93 (original numeration). Rationale:
1) output schema-qualified table name, not just the table name itself
2) reduce code bloating

File: trunkfingerprint--1.2.0.sql [hunk 3]
```diff
@@ -90,6 +89,13 @@ begin
...
+                                 when attname like '%pnnspid'   then '(select nspname from pg_namespace where oid = foo.' || attname || ')'
```
Comment (bashtanov) [inline trunkfingerprint--1.2.0.sql:93]: add it to line 69 above to reduce code duplication

File: trunkfingerprint--1.2.0.sql [hunk 4]
```diff
@@ -90,6 +89,13 @@ begin
...
+                                 when attname like '%pnpubid'   then '(select pubname from pg_publication where oid = foo.' || attname || ')'
+                                 when attname like '%tgparentid'
+                                                                then '(select concat(ns.nspname, ''.'' ,tg.tgname)
```
Comment (bashtanov) [inline trunkfingerprint--1.2.0.sql:96]: As far as I understand, trigger name is not guaranteed to be unique within a namespace, only within triggers of a given table. Also, concatenating like this is prone to ambiguity, as namespace or trigger name may contain a dot themselves. This is not mere pedantry, we do have objects with dots in their names, albeit not triggers: these are foreign tables in schemas like `centraldb`.

I'd suggest `select  (tg.tgrelid::regclass::text, tg.tgname)::text from ...`. Casting a tuple into text is always reversible, as PG cares about escaping etc. So is casting a table oid to text via regclass, it escapes and adds schema prefix automatically where needed.

**Review by bashtanov — COMMENTED** [review]
Thanks Rob, I've commented the ones that concerned me, others look good to me.

**Review by kali-brandwatch — APPROVED** [review]

**Review by bashtanov — APPROVED** [review]
looks good to me if tested, thanks Rob

**Pull request merged**